lib/galaxy/tool_util/deps/resolvers/__init__.py

```diff
@@ -240,9 +241,16 @@ class InstallableDependencyResolver(metaclass=ABCMeta):
     """ Mix this into a ``DependencyResolver`` and implement to indicate
     the dependency resolver can attempt to install new dependencies.
     """
+    read_only = False
```

Comment:
Given that the presence of the `install_dependency` method or inheriting from `InstallableDependencyResolver` is not going to be a reliable indication of the resolver's ability to install dependencies any more, I suggest to merge this class into `DependencyResolver`.

Also, these following 2 lines need to be updated any way to check for `read_only` instead of using `hasattr`:

https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tool_util/deps/views.py#L147 (`resolver.disabled` should probably also checked here, raising a different exception)

https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tool_util/deps/views.py#L201